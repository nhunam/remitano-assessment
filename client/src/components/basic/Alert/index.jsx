import PropTypes from "prop-types";
import React from "react";

function Alert(props) {
  const { message, type, onClick } = props;
  return (
    <div className={`alert alert-${type}`} role="alert">
      {message}
      <button
        type="button"
        className="close"
        data-dismiss="alert"
        aria-label="Close"
        onClick={onClick}
      >
        <span aria-hidden="true">&times;</span>
      </button>
    </div>
  );
}
Alert.defaultProps = {
  type: "info",
  show: false
};
Alert.propTypes = {
  message: PropTypes.string,
  type: PropTypes.string,
  onClick: PropTypes.func
};

export default Alert;
