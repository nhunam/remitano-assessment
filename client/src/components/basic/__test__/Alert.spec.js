import Alert from "components/basic/Alert";
import React from "react";
import { mount } from "enzyme";

describe("Alert", () => {
  const onClick = jest.fn();
  const wrapper = mount(<Alert message={"test"} onClick={onClick} />);
  it("should render properly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
