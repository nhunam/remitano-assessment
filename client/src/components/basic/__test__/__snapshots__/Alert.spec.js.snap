// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert should render properly 1`] = `ReactWrapper {}`;
